renpy/exports.py

```diff
@@ -926,7 +929,12 @@ def input(prompt, default='', allow=None, exclude='{}', length=None, with_none=N
     if fixed:
         renpy.ui.saybehavior()
 
-    rv = renpy.ui.interact(mouse='prompt', type="input", roll_forward=roll_forward)
+    rv = None
+    if renpy.emscripten:
+        rv = emscripten.run_script_string("(function callMobilePrompt() { if (window.ontouchstart !== undefined) { return prompt('%s', '%s'); } return ''})()" % (prompt, default))
```

Comment:
Using %s like this is bad - think about what would happen if prompt or default contained an apostrophe.

Reply:
You are right, I'll try to make it safer so it won't fail in case there's an apostrophe or something else that can fail the script. As for use of %, it might be the only way, `.format` will not work here, because the string contains a lot of `{` and `}` and it fails as a result of it.

Reply:
Maybe `"something(%r, %r)" % (prompt, default)` ? It would use repr and so quote the strings automatically.
That, or .format and doubling the four braces in the string.